consensus/safety-rules/src/simple_push_metrics.rs

```diff
@@ -0,0 +1,50 @@
+use std::{
+    sync::{
+        atomic::{AtomicI64, Ordering},
+        Arc, Mutex,
+    },
+    thread,
+    thread::JoinHandle,
+    time::Duration,
+};
+
+pub struct Counter {
+    pub counter_name: String,
+    pub counter: AtomicI64,
+}
+
+pub struct SerCounter {
+    pub counter_name: String,
+    pub counter: i64,
+}
+
+pub trait Metrics {
+    fn get_metrics(&self) -> Vec<&Counter>;
+}
+
+pub struct CountersPusher {}
+
+impl CountersPusher {
+    pub fn start(&self, metrics: Arc<dyn Metrics + Send + Sync>) -> JoinHandle<()> {
+        thread::spawn(move || loop {
+            let mut data = "".to_string();
```

Comment:
ideally this is broken into function calls so we can perform some form of integration tests